src/dbnode/storage/entry.go

```diff
@@ -141,14 +141,52 @@ func (entry *Entry) IndexedForBlockStart(indexBlockStart xtime.UnixNano) bool {
 	return isIndexed
 }
 
-// IndexedRange returns minimum and maximum blockStart values covered by index entry.
+// ReconciledIndexedForBlockStart returns true if the blockStart has been indexed.
+// Attempts to retrieve the most recent index entry from the shard if this entry
+// was never inserted there. If there is an error, returns the value for this entry.
+// Additionally, returns a boolean to denote if result came from reconciliation.
+func (entry *Entry) ReconciledIndexedForBlockStart(indexBlockStart xtime.UnixNano) (bool, bool) {
+	if entry.insertTime.Load() > 0 {
+		return entry.IndexedForBlockStart(indexBlockStart), false
+	}
+	e, _, err := entry.Shard.TryRetrieveSeriesAndIncrementReaderWriterCount(entry.ID)
+	if err != nil || e == nil {
+		return entry.IndexedForBlockStart(indexBlockStart), false
+	}
+	defer e.DecrementReaderWriterCount()
+
+	return e.IndexedForBlockStart(indexBlockStart), true
+}
+
+// ReconciledIndexedRange returns minimum and maximum blockStart values covered by index entry.
 // The range is inclusive. Note that there may be uncovered gaps within the range.
-// Returns (0, 0) for an empty range.
-func (entry *Entry) IndexedRange() (xtime.UnixNano, xtime.UnixNano) {
-	entry.reverseIndex.RLock()
-	min, max := entry.reverseIndex.indexedRangeWithRLock()
-	entry.reverseIndex.RUnlock()
-	return min, max
+// Attempts to retrieve the most recent index entry from the shard if this entry
+// was never inserted there. If there is an error, returns the range on this entry.
+// Returns (0, 0) for an empty range. Additionally, returns a boolean to denote
+// if result came from reconciliation.
+func (entry *Entry) ReconciledIndexedRange() (xtime.UnixNano, xtime.UnixNano, bool) {
+	var (
+		min, max   xtime.UnixNano
+		reconciled bool
+	)
+	if entry.insertTime.Load() > 0 {
+		min, max = entry.reverseIndex.indexedRangeWithRLock()
+	} else {
```

Comment:
nit: Can probably early return here yeah? 

```suggestion
	if entry.insertTime.Load() > 0 {
		min, max = entry.reverseIndex.indexedRangeWithRLock()
		return min, max, false
	}
```